### PR TITLE
Updated licence flow to use new models

### PR DESCRIPTION
**Pull request by joseluisgraa**

### CODE REVIEW

File: src/server/common/model/section/licence.js [hunk 1]
```diff
@@ -9,8 +10,10 @@ import { EmailAddress } from '../answer/email-address.js'
  */
 
 export class Licence extends SectionModel {
+  firstPage = new EmailAddressPage()
+
   get emailAddress() {
-    return this._data?.emailAddress
+    return this._data?.emailAddress.answer
   }
```
[the remark this answers is not in the document]
Comment (joseluisgraa) [reply src/server/common/model/section/licence.js:17]: I think it's used in the licence/check-answers template when we do:
```
value: {
          text: licence.emailAddress.html
        },
```
where we get the emailAddress using the getter then the html from it

[the remark this answers is not in the document]
Comment (hughfdjackson) [reply src/server/common/model/section/licence.js:17]: aah - okay.  Then let's wait to remove this until we have some way of sharing the template / view model generation.

File: src/server/common/model/section/licence.js [hunk 2]
```diff
@@ -19,7 +22,10 @@ export class Licence extends SectionModel {
    */
   static fromState(state) {
     return new Licence({
-      emailAddress: EmailAddress.fromState(state?.emailAddress)
+      emailAddress: {
+        page: new EmailAddressPage(),
+        answer: EmailAddress.fromState(state?.emailAddress)
```
[the remark this answers is not in the document]
Comment (joseluisgraa) [reply src/server/common/model/section/licence.js:27]: you mean this?

`static fromState(state) {
    const page = new EmailAddressPage()
    return new Licence({
      emailAddress: {
        page,
        answer: page.Answer.fromState(state?.emailAddress)
      }
    })
  }`

[the remark this answers is not in the document]
Comment (hughfdjackson) [reply src/server/common/model/section/licence.js:27]: No need to instantiate it - we already have an exported instance in the page.

[the remark this answers is not in the document]
Comment (hughfdjackson) [reply src/server/common/model/section/licence.js:27]: See this commit for an example: https://github.com/DEFRA/apha-apps-perms-move-animal-ui/pull/71/commits/083027f91e0262d02b1fe03b08538673f075889c

**Pull request merged**